core/src/main/java/bisq/core/notifications/MobileModel.java

```diff
@@ -131,9 +139,15 @@ boolean parseDescriptor(String descriptor) {
                 String model = descriptorTokens[0];
                 if (model.equals("iPhone")) {
                     String versionString = descriptorTokens[1];
-                    versionString = versionString.substring(0, 1);
-                    if (versionString.equals("X") || versionString.equals("SE"))
+                    String[] validVersions = {"X", "XS", "XR"};
+                    if (Arrays.asList(validVersions).contains(versionString)) {
                         return true;
+                    }
+                    if (versionString.matches("\\d[^\\d]")) {
+                        versionString = versionString.substring(0, 1);
+                    } else if (versionString.matches("\\d{2}[^\\d]")) {
+                        versionString = versionString.substring(0, 2);
+                    }
                     try {
                         int version = Integer.parseInt(versionString);
                         return version > 5;
```

Comment:
One thing I would like clarification on, if anyone knows...

From the comment on [line 134](https://github.com/bisq-network/bisq/pull/3277/files#diff-975a294e288c01124c0bb32007568ffcR134):
>         // iPhone 6 does not support isContentAvailable, iPhone 7 does.
>         // We don't know for other versions, but lets assume all above iPhone 6 are ok.

If that is the case, I assume this should be changed to:
`return version > 6;`

Unless the comment is incorrect?

I am not familiar with isContentAvailable, but is this capability really device dependent?
It appears content-available was added to notifications in iOS7: https://stackoverflow.com/questions/20690220/proper-use-of-content-available-in-ios-7-push-notifications
And iOS7 is supported as far back as iPhone 4: http://osxdaily.com/2013/06/10/ios-7-supported-devices/

So then maybe it should be:
`return version >= 4;`

Reply:
Yes, I looks like as if the check is wrong. Even if the hardware string is used it must be even higher (iPhone9,1+ == iPhone 7). Maybe @joachimneumann could shed some light on this.

Reply:
Hi,

you are probably right. I have copied the file MobileModel.java from somewhere on the Internet and also, I have not spend a lot of time to identify the hardware and software requirements. I only remember that the notifications did not work on an iPhone 6, but they did work on an iPhone 6S. Since newer operating systems (still true for iOS 13) run on the iPhone 6S, but not on the iPhone 6 and I assumed that most user will update their OS, I opted for the lazy route.

If you have older hardware and/or iPhones with older OS versions to test this, feel free to correct the requirements.

Joachim

Reply:
Thanks! I have updated this PR to reflect isContentAvailable supports iPhone 6s and newer.